### PR TITLE
Test coverage - github action

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

File: .github/workflows/test_coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+      - name: Compile
+        shell: bash
+        run: |
+          # Manually add QT executables to path
+          export PATH=/opt/qt515/bin:$PATH
+          /opt/qt515/bin/qmake QMAKE_CXX=clang++ QMAKE_LINK=clang++ CONFIG+=debug
```
Comment (flodolo) [inline .github/workflows/test_coverage.yaml]: Just `qmake` will work, since it's in path thanks to the line above.

**Comment by flodolo** [comment]
Sorry, I messed up locally with my branches. I was trying to test it on my branch, and ended up force pushing here 🤦🏼

I'll revert the change.

**Comment by bakulf** [comment]
I rebased the patch on top of the latest main.

File: .github/workflows/test_coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Unit Test Coverage
+on:
+  push:
+    branches:
+      - master
```
Comment (flodolo) [inline .github/workflows/test_coverage.yaml:5]: main

Are you sure you want this also on pushes, and not just pull requests? One alternative approach is to block direct commits to main, and force everyone to go through pull requests.

File: .github/workflows/test_coverage.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  pull_request:
+    branches:
+      - master
```
Comment (flodolo) [inline .github/workflows/test_coverage.yaml:8]: main

**Review by flodolo — APPROVED** [review]
> Sorry, I messed up locally with my branches. I was trying to test it on my branch, and ended up force pushing here 🤦🏼

Still wondering why that happened, given I have origin pointing to my fork 🙄 

Anyhow, I tested it on my branch with the modifications and it runs successfully.

File: .github/workflows/test_coverage.yaml (hunk 2 above)
Comment (bakulf) [reply .github/workflows/test_coverage.yaml]: eventually, the blocking of commits is what I want, but there I need to find a C++ reviewer first.

**Pull request merged**